src/gui/MainWindow.cpp

```diff
@@ -464,6 +464,11 @@ void MainWindow::setMenuActionState(DatabaseWidget::Mode mode)
             m_ui->actionDatabaseMerge->setEnabled(m_ui->tabWidget->currentIndex() != -1);
 
             m_searchWidgetAction->setEnabled(true);
+
+            //if not already in search mode focus the search widget and select all its content
+            if (!inSearch) {
+                m_ui->toolBar->findChild<SearchWidget *>("SearchWidget")->searchFocus();
```

Comment:
Typically you do not want to invoke a hard connection to a slot since it creates a strong dependency. Instead you should connect a signal (such as database unlock signal) to the SearchWidget's searchFocus slot. I believe this can be accomplished directly in SearchWidget.cpp.